### ISSUE TITLE
v1.14.11 CPU load issue

### ISSUE THREAD

**Issue by pleunv** [remy/nodemon#1225]
- `nodemon -v`: 1.14.11
- `node -v`: v8.9.1
- Operating system/terminal environment: Windows 10 (cmder/ConEmu)
- Command you ran: `nodemon scripts/bin/server.js --ignore dist --ignore coverage --ignore tests --ignore src --ignore cypress`

### Expected behaviour

Nodemon launches server, afterwards the process' CPU usage drops to ~0%.

### Actual behaviour

Nodemon launches server but its process is permanently using up a full CPU core

### Steps to reproduce

/

I've verified that this behavior only starts manifesting in 1.4.11, 1.4.10 does not suffer from this.
If there's anything else you want me to provide you, please let me know.

Dump: 

```
--------------
node: v8.9.1
nodemon: 1.14.11
command: C:\Tools\nodejs\node.exe C:\Projects\<PROJECT_DIR>\node_modules\nodemon\bin\nodemon.js scripts/bin/server.js --ignore dist --ignore coverage --ignore tests --ignore src --dump
cwd: C:\Projects\<PROJECT_DIR>
OS: win32 x64
--------------
{ run: false,
  system:
   { cwd: 'C:\\Projects\\<PROJECT_DIR>' },
  required: false,
  dirs:
   [ 'C:\\Projects\\<PROJECT_DIR>' ],
  timeout: 1000,
  options:
   { ignore:
      [ '**/.git/**',
        '**/.nyc_output/**',
        '**/.sass-cache/**',
        '**/bower_components/**',
        '**/coverage/**',
        '**/node_modules/**',
        'dist',
        'coverage',
        'tests',
        'src',
        re: /.*.*\/\.git\/.*.*|.*.*\/\.nyc_output\/.*.*|.*.*\/\.sass\-cache\/.*.*|.*.*\/bower_components\/.*.*|.*.*\/coverage\/.*.*|.*.*\/node_modules\/.*.*|dist|coverage|tests|src/ ],
     dump: true,
     watch: [ '*.*', re: /.*\..*/ ],
     ignoreRoot:
      [ '**/.git/**',
        '**/.nyc_output/**',
        '**/.sass-cache/**',
        '**/bower_components/**',
        '**/coverage/**',
        '**/node_modules/**' ],
     restartable: 'rs',
     colours: true,
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     runOnChangeOnly: false,
     verbose: false,
     signal: 'SIGUSR2',
     stdout: true,
     watchOptions: {},
     execOptions:
      { script: 'scripts/bin/server.js',
        exec: 'node',
        args: [],
        scriptPosition: 0,
        nodeArgs: undefined,
        execArgs: [],
        ext: 'js,mjs,json',
        env: {} },
     monitor:
      [ '*.*',
        '!**/.git/**',
        '!**/.nyc_output/**',
        '!**/.sass-cache/**',
        '!**/bower_components/**',
        '!**/coverage/**',
        '!**/node_modules/**',
        '!C:\\Projects\\<PROJECT_DIR>\\dist/**/*',
        '!C:\\Projects\\<PROJECT_DIR>\\coverage/**/*',
        '!tests',
        '!C:\\Projects\\<PROJECT_DIR>\\src/**/*' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [],
  watchInterval: null,
  signal: 'SIGUSR2',
  command:
   { raw: { executable: 'node', args: [ 'scripts/bin/server.js' ] },
     string: 'node scripts/bin/server.js' } }
--------------
```

**Comment by remy** [comment]
Can you give me the initial output of nodemon using the `-V` flag?

**Comment by pleunv** [comment]
```
to be executed: "nodemon scripts/bin/server.js --ignore dist --ignore coverage --ignore tests --ignore src -V"
[nodemon] 1.14.11
[nodemon] to restart at any time, enter `rs`
[nodemon] or send SIGHUP to 19000 to restart
[nodemon] ignoring: .\dist/**/* .\coverage/**/* tests .\src/**/*
[nodemon] watching: *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node scripts/bin/server.js`
[nodemon] forking
[nodemon] child pid: 3332
```

`/tests` is a directory that does not seem to exist anymore, which is why it is not globbed I guess. Leaving it out makes no difference.

**Comment by pleunv** [comment]
Running it on `1.14.10` gives me the actual file count at the bottom, not getting this on `1.14.11`:

```
to be executed: "nodemon scripts/bin/server.js --ignore dist --ignore coverage --ignore src -V"
[nodemon] 1.14.10
[nodemon] to restart at any time, enter `rs`
[nodemon] or send SIGHUP to 10496 to restart
[nodemon] ignoring: .\dist/**/* .\coverage/**/* .\src/**/*
[nodemon] watching: *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node scripts/bin/server.js`
[nodemon] forking
[nodemon] child pid: 18608
[nodemon] watching 16 files
```

**Comment by remy** [comment]
Hmm, I suspect the CPU hogging is nodemon actually counting up the files… 

I'm investigating.

**Comment by remy** [comment]
I'm pretty confident this is a windows (perhaps cmdr) specific. Running cmdr locally on a near empty directory yields the following:

```bash
C:\Users\Demo\nodemon-windows  (nodemon-windows@1.0.0)
λ ll
'll' is not recognized as an internal or external command,
operable program or batch file.

C:\Users\Demo\nodemon-windows  (nodemon-windows@1.0.0)
λ ls
http.js  node_modules/  package-lock.json  package.json

C:\Users\Demo\nodemon-windows  (nodemon-windows@1.0.0)
λ nodemon -V http.js
[nodemon] 1.14.11
[nodemon] to restart at any time, enter `rs`
[nodemon] or send SIGHUP to 1188 to restart
[nodemon] watching: *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `node http.js`
[nodemon] forking
[nodemon] child pid: 13744
this was added
ok.
[nodemon] watching 1302 files
```

Note the **1302** files - I suspect it's going in an walking the node_modules directory.

**Comment by pleunv** [comment]
You're right, adding `--ignore node_modules` seems to work around this issue. Never occurred to me to try this yesterday :).

Fwiw, native windows prompt has the same problem, so it's not limited to cmder/ConEmu.

**Comment by remy** [comment]
Found the [related issue](https://github.com/paulmillr/chokidar/issues/668) on Chokidar and I'm about to send a PR to fix.

**Comment by pleunv** [comment]
Awesome, thank you!

**Comment by remy** [comment]
Just to let you know, I'm still waiting for the chokidar 2.0.1 release to land. My change got merged in, but they're still holding back the release (I'm assuming for other patches to settle first).

**Comment by remy** [comment]
Still waiting on Chokidar

**Comment by lensbart** [comment]
I’m having the same (or a very similar) issue on macOS 10.13.3 — the node process stays at around 100% CPU for a prolonged time, after which it eventually drops to normal levels. My guess was that this would be due to indexing — perhaps `node_modules` — but the documentation says that `node_modules` is excluded by default, and including `--ignore node_modules` as suggested above doesn’t seem to work.

**Comment by lensbart** [comment]
@remy it seems to be a different issue still, as I’m not specifying a `--watch` option as is the case in the suggested link.

- `nodemon -v`: 1.14.12
- `node -v`: v8.9.4
- macOS 10.13.3
- tried commands `nodemon --ignore ./node_modules ./build/app.js`, `nodemon --ignore ../node_modules ./build/app.js`, `nodemon --ignore node_modules ./build/app.js` and `nodemon --ignore node_modules/ ./build/app.js`

**Comment by remy** [comment]
@lensbart cheers for those, I've got a _very_ similar environment, so I'll setup a local test to see what's going on. It should _definitely_ be ignoring the `node_modules` directories. I might file a new issue for this if there's something going on (again, the issue we're commenting on here is directly linked to use on Window).

**Comment by Dosflange** [comment]
This is a very old thread, I am also experiencing this (100% CPU load) running nodemon 3.1.4 (Archlinux on a Intel® Core™ i7-10700T). I only noticed this because I have a passively-cooled system, and the case radiated heat across my desk running one 20k static website. I was not using Sublime...
